src/water_map.h

```diff
@@ -78,6 +78,16 @@ enum LockPart {
 
 bool IsPossibleDockingTile(Tile t);
 
+/**
+ * Get the type of water tile: clear, lock or depot.
+ * @param t Water tile to query.
+ * @return WBL_TYPE_NORMAL, WBL_TYPE_LOCK or WBL_TYPE_DEPOT.
+ */
+static inline WaterTileTypeBitLayout GetWaterTileClass(Tile t) {
+	assert(IsTileType(t, MP_WATER));
+	return (WaterTileTypeBitLayout)GB(t.m5(), WBL_TYPE_BEGIN, WBL_TYPE_COUNT);
```

Comment:
We are moving to static casts instead of C-style casts. 🙂 

```suggestion
	return static_cast<WaterTileTypeBitLayout>(GB(t.m5(), WBL_TYPE_BEGIN, WBL_TYPE_COUNT));
```